src/sentry/api/endpoints/project_filter_details.py

```diff
@@ -80,21 +80,12 @@ def put(self, request: Request, project, filter_id) -> Response:
 
         returned_state = None
         if filter_id == "legacy-browsers":
-            if isinstance(current_state, bool) or isinstance(new_state, bool):
-                returned_state = new_state
-                if not new_state:
-                    audit_log_state = audit_log.get_event_id("PROJECT_DISABLE")
+            returned_state = new_state
 
-            elif current_state - new_state:
-                returned_state = current_state - new_state
+            # Check if disabling filter
+            if not new_state:
                 audit_log_state = audit_log.get_event_id("PROJECT_DISABLE")
 
-            elif new_state - current_state:
-                returned_state = new_state - current_state
-
-            elif new_state == current_state:
-                returned_state = new_state
-
```

Comment:
This numerical logic can be ignored because `new_state` and `current_state` cannot be numbers for the legacy-browser filter.

Instead, we now always set `returned_state` to `new_state` and add a check to see if we audit log should say disabled.